src/handler.go

```diff
@@ -201,46 +244,97 @@ func CreateTripEntryHandler(TripCtl controller.TripCtl) gin.HandlerFunc {
-		// TO-DO
 		ctx := c.Request.Context()
 
-		response, err := TripCtl.UpdateTripEntry(ctx, nil)
+		tripIdParam := c.Param(model.ExpenseParamTripId)
+		tripId, err := uuid.Parse(tripIdParam)
 		if err != nil {
-			c.JSON(http.StatusInternalServerError, err.Error())
+			utils.HandleErrorAndAbort(c, *expenseerror.EXPENSE_BAD_REQUEST)
+			return
+		}
+
+		var tripUpdateRequest model.TripUpdateRequest
+		if err := c.ShouldBindJSON(&tripUpdateRequest); err != nil {
+			utils.HandleErrorAndAbort(c, *expenseerror.EXPENSE_BAD_REQUEST)
+			return
+		}
+
+		response, serviceErr := TripCtl.UpdateTripEntry(ctx, &tripId, tripUpdateRequest)
+		if serviceErr != nil {
+			utils.HandleErrorAndAbort(c, *serviceErr)
 			return
 		}
 
 		c.JSON(http.StatusOK, response)
 	}
 }
 
-func GetTripDetailsHandler(TripCtl controller.TripCtl) gin.HandlerFunc {
+func DeleteTripEntryHandler(TripCtl controller.TripCtl) gin.HandlerFunc {
 	return func(c *gin.Context) {
-		// TO-DO
 		ctx := c.Request.Context()
 
-		response, err := TripCtl.GetTripDetails(ctx, nil)
+		tripIdParam := c.Param(model.ExpenseParamTripId)
+		tripId, err := uuid.Parse(tripIdParam)
 		if err != nil {
-			c.JSON(http.StatusInternalServerError, err.Error())
+			utils.HandleErrorAndAbort(c, *expenseerror.EXPENSE_BAD_REQUEST)
+			return
+		}
+
+		serviceErr := TripCtl.DeleteTripEntry(ctx, &tripId)
+		if serviceErr != nil {
+			utils.HandleErrorAndAbort(c, *serviceErr)
+			return
+		}
+
+		c.JSON(http.StatusNoContent, nil)
+	}
+}
+
+func InviteUserToTripHandler(TripCtl controller.TripCtl) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		ctx := c.Request.Context()
+
+		tripIdParam := c.Param(model.ExpenseParamTripId)
+		tripId, err := uuid.Parse(tripIdParam)
+		if err != nil {
+			utils.HandleErrorAndAbort(c, *expenseerror.EXPENSE_BAD_REQUEST)
+			return
+		}
+
+		var inviteUserRequest model.InviteUserRequest
+		if err := c.ShouldBindJSON(&inviteUserRequest); err != nil {
+			utils.HandleErrorAndAbort(c, *expenseerror.EXPENSE_BAD_REQUEST)
+			return
+		}
+
+		response, serviceErr := TripCtl.InviteUserToTrip(ctx, &tripId, inviteUserRequest)
+		if serviceErr != nil {
+			utils.HandleErrorAndAbort(c, *serviceErr)
 			return
 		}
 
 		c.JSON(http.StatusOK, response)
 	}
 }
 
-func DeleteTripEntryHandler(TripCtl controller.TripCtl) gin.HandlerFunc {
+func AcceptTripInviteHandler(TripCtl controller.TripCtl) gin.HandlerFunc {
 	return func(c *gin.Context) {
-		// TO-DO
 		ctx := c.Request.Context()
 
-		err := TripCtl.DeleteTripEntry(ctx, nil)
+		tripIdParam := c.Param(model.ExpenseParamTripId)
+		tripId, err := uuid.Parse(tripIdParam)
 		if err != nil {
-			c.JSON(http.StatusInternalServerError, err.Error())
+			utils.HandleErrorAndAbort(c, *expenseerror.EXPENSE_BAD_REQUEST)
 			return
 		}
 
-		c.JSON(http.StatusOK, nil)
+		serviceErr := TripCtl.AcceptTripInvite(ctx, &tripId)
+		if serviceErr != nil {
+			utils.HandleErrorAndAbort(c, *serviceErr)
+			return
+		}
+
+		c.JSON(http.StatusNoContent, nil)
```

Comment:
Sollte das nicht lieber `200 OK` sein?

Reply:
Für HTTP-Antworten ohne Content ist `204 NO CONTENT` geeignet, laut Best Practices.

Reply:
Laut [Spec](https://www.rfc-editor.org/rfc/rfc2616#section-10.2.5) sollte 204 genutzt werden, wenn du der `User Agent` nicht weiternavigieren soll. Ist bei uns nicht der Fall, da der User danach im Best Case auf die Trip-Details Page weitergeleitet werden soll. Würde also `200` als angemessener sehen.

Reply:
Der HTTP-Statuscode 204 gibt nur an, dass die Anfrage erfolgreich war und es keine zusätzlichen Daten zum Senden gibt. Er hat nichts mit der Navigation oder Weiterleitung auf der Clientseite zu tun. 
Ich kann es aber gerne bei 200 lassen wenn es Dir besser passt ^^

Reply:
Steht doch in der RFC-Spec drin, die ich verlinkt hab?

> If the client is a user agent, it SHOULD NOT change its document view
>    from that which caused the request to be sent. This response is
>    primarily intended to allow input for actions to take place without
>    causing a change to the user agent's active document view, although
>    any new or updated metainformation SHOULD be applied to the document
>    currently in the user agent's active view.